File: leetcode/problems/java/group-anagram.java
import java.util.*;

class Solution {
   public List<List<String>> groupAnagrams(String[] strs) {
         HashMap<String, List<String>> map = new HashMap<>();
         for (String str : strs) {
               char[] charArray = str.toCharArray();
               Arrays.sort(charArray);
               String sortedStr = new String(charArray);
               if (!map.containsKey(sortedStr)) {
                  map.put(sortedStr, new ArrayList<>());
               }
               map.get(sortedStr).add(str);
         }
         return new ArrayList<>(map.values());
   }
}

File: leetcode/problems/rs/group-anagram.rs
pub struct Solution;

impl Solution {
   pub fn group_anagrams(strs: Vec<String>) -> Vec<Vec<String>> {
      use std::collections::HashMap;
      let mut map: HashMap<Vec<u8>, Vec<String>> = HashMap::new();
      for s in strs {
         let mut key = s.as_bytes().to_vec();
         key.sort();
         map.entry(key).or_insert(Vec::new()).push(s);
      }
      map.into_iter().map(|(_, v)| v).collect()
   }
}
